core/transformer/src/PendingReferenceMap.ts

```diff
@@ -0,0 +1,119 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module Utils
+ */
+
+import { Id64String } from "@itwin/core-bentley";
+
+interface PendingReferenceProps {
+  referencer: Id64String;
+  referenced: Id64String;
+  isModelRef: boolean;
+}
+
+/**
+ * A reference relationships from an element, "referencer", to an element or its submodel, "referenced"
+ * @internal
+ */
+export class PendingReference implements PendingReferenceProps {
+  public referencer!: Id64String;
```

Comment:
What is motivation for non-null assertions on members and `Object.assign` in constructor vs just assigning to properties directly in constructor?

Reply:
it's a stylistic choice, that I don't need to enumerate the properties twice. If it's unprecedented in our other props-constructed types I can go to the assignment style

Reply:
Yeah, it's non-idiomatic. It also will copy any other unrelated properties that may be present on the input.

Reply:
this was eliminated while resolving the following comment, by removing the class entirely and just operating on the object